aws/onboarding/byok8s/python/rubrik_eks_cluster_private.py

```diff
@@ -0,0 +1,1203 @@
+                    }
+                }
+            }]
+        }
+
+        config_text = yaml.dump(cluster_config, default_flow_style=False)
+
+        # Determine the path for the kubectl config file
+        kube_dir = expanduser("~") + "/.kube"
+        config_file = kube_dir + "/config"
+
+        # Ensure the .kube directory exists
+        if not os.path.exists(kube_dir):
+            print(f"Directory {kube_dir} does not exist. Creating it.")
+            os.makedirs(kube_dir)
+
+        print("Writing kubectl configuration to ", config_file)
+        with open(config_file, "w") as file:
+            file.write(config_text)
+        print("Done: Written kubectl configuration to ", config_file)
+        return config_file
+    except Exception as e:
+        print(f"Error persisting kubectl configuration: {str(e)}")
+        raise
+
+
+def get_config_map(
+        worker_node_role_arn: str, cross_account_role_arn: str
+) -> client.V1ConfigMap:
+    """Setup config map for connecting worker nodes to EKS cluster
+
+        Args:
+        Returns:
+            returns config map
+    """
+    try:
+        worker_node_role_arn_without_path = strip_paths_from_arn(
+            worker_node_role_arn
+        )
+        cross_account_role_arn_without_path = strip_paths_from_arn(
+            cross_account_role_arn
+        )
+
+        config_map = client.V1ConfigMap(
+            api_version="v1",
+            kind="ConfigMap",
+            metadata=client.V1ObjectMeta(
+                name=CONFIG_MAP_NAME,
+                namespace=KUBE_SYSTEM_NAMESPACE,
+            ),
+            data={
+                MAP_ROLES_DATA_KEY:
+                    MAP_ROLE_DATA_FORMAT % (
+                        worker_node_role_arn_without_path,
+                        cross_account_role_arn_without_path,
+                    ),
+            },
+        )
+        return config_map
+    except Exception as e:
+        print(f"Error getting config map: {str(e)}")
+        raise
+
+
+def connect_worker_nodes_to_cluster(
+        worker_node_role_arn: str,
+        cross_account_role_arn: str,
+        cluster_endpoint: str,
+        cluster_certificate_authority: str,
+        cluster_name: str,
+        cluster_arn: str
+) -> None:
+    """Connect worker nodes to EKS cluster.
+
+    As part of this we create a config map in kube-system namespace with the
+    worker node role ARN and cross account role ARN. This config map is used by
+    the aws-auth controller to map the IAM roles to Kubernetes users and groups.
+
+        Args:
+            worker_node_role_arn: worker node role ARN
+            cross_account_role_arn: cross account role ARN
+            cluster_endpoint: cluster endpoint
+            cluster_certificate_authority: cluster certificate authority
+            cluster_name: cluster name
+            cluster_arn: cluster ARN
+        Returns:
+            None
+    """
+    try:
+        config_path = persist_kubectl_configuration(
```

Comment:
nit: should we also delete the kubeconfig file once the config map has been applied?

Reply:
If we delete, we won't be able to access the cluster from this jumpbox. Then we would need to reconfigure again.

Reply:
Ok